lib/Serialization/Deserialization.cpp

```diff
@@ -140,15 +140,27 @@ void ExtensionError::anchor() {}
 /// Returns true if the next entry is a record of type \p recordKind.
 /// Destroys the stream position if the next entry is not a record.
 static void skipRecord(llvm::BitstreamCursor &cursor, unsigned recordKind) {
-  auto next = cursor.advance(AF_DontPopBlockAtEnd);
+  Expected<llvm::BitstreamEntry> maybeNext =
+      cursor.advance(AF_DontPopBlockAtEnd);
+  if (!maybeNext) {
+    // FIXME this drops the error on the floor.
+    consumeError(maybeNext.takeError());
```

Comment:
This should probably at least assert, to match the current behavior.

Reply:
I made it `cantFail`.

Reply:
That turns out to not be something we can do at the moment because it causes failures. I'll use bug-compatible old behavior.

Reply:
Uh. Are you saying we have existing corrupted modules?

Reply:
I haven't dug in :)